src/kuzzle.js

```diff
@@ -434,6 +434,11 @@ Kuzzle.prototype.login = function (strategy, credentials, expiresIn, cb) {
       strategy: strategy
     };
 
+  // don't need credentials in case of an OAUTH strategy
+  if (typeof credentials === 'number' || typeof credentials === 'string') {
```

Comment:
Does not work if this method is called with only a strategy and a callback.
I suggest prototyping this function only with the `strategy` argument, and then going through the `arguments` global variable to assign other arguments depending on their type.